src/mon/MonClient.cc

```diff
@@ -400,6 +406,17 @@ void MonClient::flush_log()
   send_log();
 }
 
+void MonClient::handle_quorum_peon(MMonMap *m) {
+  auto con_addrs = m->get_source_addrs();
+  auto mon_name = monmap.get_name(con_addrs);
+
+  MonMap peon_map;
+  auto p = m->monmapbl.cbegin();
```

Comment:
What's the intention behind doing this outside the message's dencoding methods?

Reply:
i matched it to the handle_monmap method. the monmapbl is a bufferlist that still needs another decode to become MonMap, as I understand from the original handle_monmap method.